### ISSUE TITLE
[mg26x] Upgrade mg5_amc to 2.6.0

### ISSUE THREAD

**Issue by bendavid** [cms-sw/genproductions#1290]
The new 2.6.0 release is out containing most importantly the bias weighting functionality at NLO.
(Also interesting are the finite quark mass corrections for gluon fusion production.)

Since this is closely related to developments in the 2.5 branch (and was originally supposed to be called 2.5.6) I think we should just update the mg25x genproductions branch rather than creating a new one.

**Comment by perrozzi** [comment]
for the moment a new branch is created https://github.com/cms-sw/genproductions/tree/mg26x

**Comment by perrozzi** [comment]
First thing to do is to merge mg26x into mg25x. @kdlong can you please do it asap?

**Comment by kdlong** [comment]
@perrozzi: so we'd like to drop the 2.5.5 and instead support only 2.6? 

@bendavid, do you have any issues with the 2.6 branch as it stands?

**Comment by bendavid** [comment]
As far as I know the only outstanding issue is validation.

2.6.0 is no more or less validated than 2.5.5, and in any case does not really have major changes on top.

I see no reason to support or spend any more time testing or validating 2.5.x (and I was not in favour of creating a new branch for 2.6.x in the first place.)

**Comment by perrozzi** [comment]
mg25x and mg26x are in sync https://github.com/cms-sw/genproductions/compare/mg26x...cms-sw:mg25x
we can delete mg25x if/when you give a green light

**Comment by perrozzi** [comment]
mg25x deleted

**Comment by kdlong** [comment]
This can be closed.

**Comment by perrozzi** [comment]
Hi all (@bendavid, @kdlong, @qliphy, @agrohsje, @AndreasAlbert  in particular)
as agreed yesterday during the coordination meeting we seem to be ready to merge the mg26x branch (2.6.0) to master, as is fully validated now.

After that we can merge https://github.com/cms-sw/genproductions/pull/1725 into mg26x and finalize the update to 2.6.1.

Everybody up? Comments? Questions? Suggestions?
Thanks, Efe and Luca

**Comment by AndreasAlbert** [comment]
@perrozzi works for me

**Comment by perrozzi** [comment]
looking at https://github.com/cms-sw/genproductions/compare/master...cms-sw:mg26x things are not totally trivial and some work will be needed.
Can you remind me right away if at least the cards look the same between 2.4.2 and 2.6.0 ?

**Comment by efeyazgan** [comment]
runcmsgrid.sh needs to be patched too for 2.6.0. @kdlong, @agrohsje, could you please do that as well before we merge?
Thanks, 
Luca and Efe

**Comment by agrohsje** [comment]
runcmsgrid_LO.sh and runcmsgrid_NLO.sh run fine in 2.6.0. What needs to be adjusted?

**Comment by efeyazgan** [comment]
Yes, it has runcmsgrid.sh, so the patch should be added.

**Comment by bendavid** [comment]
Forcing the number of events is NOT needed for NLO, since the code issue which leads to this situation does not exist there.

**Comment by efeyazgan** [comment]
Yes but I am talking about the LO one. The patch is not there.

**Comment by AndreasAlbert** [comment]
Is there anything preventing us from just porting the fix?
I think there is an upstream fix already included in 2.6.1, but for 2.6.0 it should be fine to use the workaround we have.

**Comment by kdlong** [comment]
No, we just need to copy it over, which should hopefully be trivial. I hope to look at it later today.

**Comment by AndreasAlbert** [comment]
PR here: https://github.com/cms-sw/genproductions/pull/1744

**Comment by perrozzi** [comment]
partial merge here https://github.com/cms-sw/genproductions/pull/1751/files
6 main conflicts to be solved, would like you to have a look.
all the rest should be less critical but still to be crosschecked

**Comment by perrozzi** [comment]
I guess this can be closed, right?

**Comment by qliphy** [comment]
Thanks @perrozzi     Looks so to me.  

 Any other comment from others?  Probably we can discuss and close this at Thursday's GEN coordination meeting.
